### PR TITLE
Add Regions for Uruguay.

### LINKED ISSUE #29729: [Issue] Add Regions for Uruguay.
### Preconditions (*)
1. Magento 2.4

### Steps to reproduce (*)
Regions for Uruguay will be added with this PR. Regions are using the ISO 3166-2 code (the regions table can be check here [https://en.wikipedia.org/wiki/ISO_3166-2:UY](https://en.wikipedia.org/wiki/ISO_3166-2:UY))

### Expected result (*)
Once you finish the installation the country should show you Regions on shipping estimation, checkout or when you edit the customer address form.

### Actual result (*)
No region for Uruguay

### PR DESCRIPTION
**Pull request by barbanet**
### Description (*)
Regions for Uruguay will be added with this PR. Regions are using the ISO 3166-2 code (the regions table can be check here [https://en.wikipedia.org/wiki/ISO_3166-2:UY](https://en.wikipedia.org/wiki/ISO_3166-2:UY))

### Manual testing scenarios (*)
1. Once you finish the installation the country should show you Regions on shipping estimation, checkout or when you edit the customer address form.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29729: Add Regions for Uruguay.

### CODE REVIEW

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            $this->getDataForUruguay()
+        );
+    }
+
+    /**
+     * Uruguay states data.
+     *
+     * @return array
+     */
+    private function getDataForUruguay()
```
Comment (BarnyShergold) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php]: Missing return type on prototype

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            ['UY', 'UY-SO', 'Soriano'],
+            ['UY', 'UY-TA', 'Tacuarembó'],
+            ['UY', 'UY-TT', 'Treinta y Tres']
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function getDependencies()
```
Comment (BarnyShergold) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php:90]: Missing return type on prototype

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    {
+        return [
+            InitializeDirectoryData::class,
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getAliases()
```
Comment (BarnyShergold) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php:100]: Missing return type on prototype

**Comment by BarnyShergold** [comment]
@magento run all tests

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+declare(strict_types=1);
+
+namespace Magento\Directory\Setup\Patch\Data;
+
+use Magento\Directory\Setup\DataInstaller;
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+
+/**
+ * Class AddDataForUruguay
```
Comment (BarnyShergold) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php]: Add better class description - has to be more than just this or tests fail

File: app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+     * @inheritdoc
+     */
+    public function apply()
+    {
+        /** @var DataInstaller $dataInstaller */
+        $dataInstaller = $this->dataInstallerFactory->create();
+        $dataInstaller->addCountryRegions(
+            $this->moduleDataSetup->getConnection(),
+            $this->getDataForUruguay()
+        );
```
Comment (BarnyShergold) [inline app/code/Magento/Directory/Setup/Patch/Data/AddDataForUruguay.php:52]: MUST return $this

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Ensure all tests are passing as well as requested changes

**Comment by sidolov** [comment]
@magento create issue

**Comment by barbanet** [comment]
Thank you @BarnyShergold for all those recommendations. I'll take care ASAP.

**Comment by engcom-Echo** [comment]
Hi, @barbanet. Could you please check failed tests and recommendations? 

Thank you.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
Hi  @barbanet. Can you fix "Static Tests"?
Failed functional tests not related to the changes in this PR

**Comment by gabrieldagama** [comment]
The risk was set to`low` due to: the suggested changes shouldn't impact other areas.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by BarnyShergold — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @BarnyShergold, thank you for the review.
[ENGCOM-8244](https://jira.corp.magento.com/browse/ENGCOM-8244) has been created to process this Pull Request
:eight_spoked_asterisk: @BarnyShergold, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
Hello @BarnyShergold 

Can you please add an appropriate label for _test coverage_?

Thank you

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
Hello @BarnyShergold

Can you please add an appropriate label for test coverage?

Thank you

**Review by BarnyShergold — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @BarnyShergold, thank you for the review.
[ENGCOM-8244](https://jira.corp.magento.com/browse/ENGCOM-8244) has been created to process this Pull Request
:eight_spoked_asterisk: @BarnyShergold, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Bravo** [comment]
:heavy_check_mark:  QA Passed

Regions are added for Uruguay country 

See the result BEFORE switching to the PR
29722_before

and AFTER switching to the PR
29722_after1

List of regions
29722_after2

Storefront
29722_storefront

:heavy_check_mark: Successfully submitted order and created shipment to Uruguay

**Pull request merged**